### DISCUSSION TITLE
PHP 8.1 support?

### DISCUSSION THREAD

**Discussion by Jameron** [phpactor/phpactor#1837] [Q&A] [answered]
Just curious if PHP 8.1 support will be added, just updated to PHP 8.1 and no longer working. Thanks.

**Comment by dantleech** [comment, marked as answer]
Works for me, any particular error?

**Reply by Jameron** [reply]
Not sure if this is something I can fix locally, but I have coonfirmed my output in command line for php --version is PHP 8.1. I removed phpactor from .vim/plugged/phpactor, and then re-installed it using PlugInstall. I then need to change directories to phpactor and run composer install, that fails with a number of errors, here are a few:


`PHP Deprecated:  Return type of Phpactor\ConfigLoader\Core\PathCandidates::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/Cameron.Macfarlane/.vim/plugged/phpactor/vendor/phpactor/config-loader/lib/Core/PathCandidates.php on line 24
Deprecated: Return type of Phpactor\ConfigLoader\Core\PathCandidates::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/Cameron.Macfarlane/.vim/plugged/phpactor/vendor/phpactor/config-loader/lib/Core/PathCandidates.php on line 24
PHP Deprecated:  Return type of Phpactor\FilePathResolver\Expanders::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/Cameron.Macfarlane/.vim/plugged/phpactor/vendor/phpactor/file-path-resolver/lib/Expanders.php on line 45`

**Reply by dantleech** [reply]
These are deprecation warnings not errors. Was there an final error which caused it to fail?

Which version of Phpactor do you have? Note that the tagged version is very out of date and you should use master for now.

**Reply by Jameron** [reply]
What is best way to change the branch for the installation?

My .vimrc looks like this: `Plug 'phpactor/phpactor', {'for': 'php', 'tag': '*', 'do': 'composer install --no-dev -o'}`

**Reply by dantleech** [reply]
Perhaps just replace `tag` with `branch` and change `*` to `master`: https://github.com/junegunn/vim-plug
